test/tests/kernels/ad_scalar_kernel_constraint/tests

```diff
@@ -0,0 +1,113 @@
+[Tests]
+  issues = '#22174'
+  design = 'source/kernels/ADScalarLMKernel.md'
+  [./kernel_dirichlet]
+    requirement = 'The system will produce a harmonic function with a prescribed average value,'
+    [physics_separate]
+      type = 'Exodiff'
+      input = 'scalar_constraint_kernel_RJ.i'
+      exodiff = 'scalar_constraint_kernel_RJ_out.e'
+      # This problem only has 4 elements and therefore does not seem to run on > 4 procs.
+      cli_args = 'Executioner/residual_and_jacobian_together=false'
+      ad_indexing_type = 'global'
+      detail = 'showing the correct results with separate computation of residual and Jacobian'
+    [../]
+    [jacobian_separate]
+      type = 'PetscJacobianTester'
+      run_sim = True
+      input = 'scalar_constraint_kernel_RJ.i'
+      ad_indexing_type = 'global'
+      detail = 'and then verifying the separated Jacobian'
```

Comment:
```suggestion
      detail = 'verifying the separated Jacobian'
```
Details are are listed as bullets under the requirement, so it typically makes sense to remove these premable/conjunction words